libc/unistd_file.c

```diff
@@ -0,0 +1,774 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * Testing unistd.h file related functions
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+#include <unity_fixture.h>
+
+#define LINE1 "line1\n"
+#define LINE2 "lineline2\n" /* longer than LINE1 */
+#define LINE3 "line3\n"     /* same length as LINE1 */
+#define LINE4 "\n"
+
+
+static const char filename[] = "unistd_file_testfile";
+static int fd, fd2;
+static char buf[50];
+
+
+TEST_GROUP(unistd_file);
+
+
+TEST_SETUP(unistd_file)
+{
+	/* open a file */
+	fd = open(filename, O_RDWR | O_CREAT | O_TRUNC);
+	TEST_ASSERT_NOT_EQUAL_INT(-1, fd);
+}
+
+
+TEST_TEAR_DOWN(unistd_file)
+{
+	memset(buf, '\0', sizeof(buf));
+	if (fd > 0)
```

Comment:
I don't think we want to close stdin, but being pedantic:
```suggestion
	if (fd >= 0)
```